posthog/api/plugin.py

```diff
@@ -0,0 +1,132 @@
+import json
+from typing import Any, Dict
+
+import requests
+from django.conf import settings
+from rest_framework import request, serializers, viewsets
+from rest_framework.decorators import action
+from rest_framework.exceptions import APIException
+from rest_framework.response import Response
+
+from posthog.models import Plugin, PluginConfig
+from posthog.plugins import download_plugin_github_zip, reload_plugins_on_workers
+
+
+class PluginSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Plugin
+        fields = ["id", "name", "description", "url", "config_schema", "tag", "from_json"]
+
+    def create(self, validated_data: Dict, *args: Any, **kwargs: Any) -> Plugin:
+        if not settings.PLUGINS_INSTALL_FROM_WEB:
+            raise APIException("Plugin installation via the web is disabled!")
+        if len(Plugin.objects.filter(name=validated_data["name"])) > 0:
```

Comment:
You could just add a DRF [UniqueValidator](https://www.django-rest-framework.org/api-guide/validators/#uniquevalidator) to the `name` field (either declaring the attribute above or using `extra_kwargs`). It will avoid the extra code and ensure the error response is standardized.